daemon/health.go

```diff
@@ -188,15 +188,26 @@ func handleProbeResult(d *Daemon, c *container.Container, result *types.Healthch
 func monitor(d *Daemon, c *container.Container, stop chan struct{}, probe probe) {
 	probeTimeout := timeoutWithDefault(c.Config.Healthcheck.Timeout, defaultProbeTimeout)
 	probeInterval := timeoutWithDefault(c.Config.Healthcheck.Interval, defaultProbeInterval)
+loop:
 	for {
 		select {
 		case <-stop:
 			logrus.Debugf("Stop healthcheck monitoring for container %s (received while idle)", c.ID)
 			return
 		case <-time.After(probeInterval):
-			logrus.Debugf("Running health check for container %s ...", c.ID)
 			startTime := time.Now()
+			healthChecksWaiters.Inc()
 			ctx, cancelProbe := context.WithTimeout(context.Background(), probeTimeout)
+			if err := d.healthcheckLimiter.Acquire(ctx, 1); err != nil {
+				// if we see this, we have ourselves a nasty bug
+				logrus.WithError(err).Warnf("Healthcheck semaphore acquisition failed", c.ID)
+				continue loop
```

Comment:
do you want to decrement here `healthChecksWaiters.Dec()` ?

Reply:
like before continue of course :)

Reply:
maybe a stats heathcheck dropped or denied something like that to capture the ones rejected